gdx/src/com/badlogic/gdx/graphics/g3d/shaders/BaseShader.java

```diff
@@ -113,6 +113,15 @@ public boolean validate (final BaseShader shader, final int inputID, final Rende
 	public Camera camera;
 	private Mesh currentMesh;
 
+	/** ClassX: thread-safety support */
+	public final Matrix4 tempMatrix4 = new Matrix4();
```

Comment:
Could this maybe be changed to `protected`?
At least for `ParticleShader` it should work (Didn't check all other occurrences).

Reply:
definitely a  good idea. We tend to not alter the fields visibility of original code.

Reply:
this matrix is referenced outside of BaseShader (i.e. ParticleShader).. so unfortunately I cannot change the visibility